 # Supplementary Information


De Anda V, Zapata PeÃ±asco I, Blaz J, Poot-Hernandez AC, Contreras Moreira B, Gonzales Laffite Marcos, Hernandez Rosales M, Gamez Tamariz N, Eguiarte Fruns E, Souza V. Understanding the mechanisms behind the response of environmental perturbations in microbial mats: a metagenomic-network based approach. Submitted in Frontiers in Microbiology 
To reproduce the figures first install [jupyter notebook](http://jupyter.org/install.html)  
and then clone the repository.  

```bash
git clone https://github.com/valdeanda/Time_series_mats.git
cd scripts 
ipython3 notebook 
```
The directory scripts contains all the steps required to compute  the main figures from the article in notebook, html and python format.

### `Libraries version`
In order to reproduce all the figures using the notebooks, the following libraries must be installed first with the corresponding versions 

+ Pandas version ==> 0.20.3 
+ Numpy ==> 1.13.3
+ Seaborn ==> 0.8.0
+ Plt ==> 1.13.3
+ Matplotlib ==> 2.02


###  Figures

<img src="https://valdeanda.github.io/Time_series_mats/figures/MEBS_mats.png" width="256" height="256" align="left">

### `Main cycles with MEBS over time`

Script available in [html](https://valdeanda.github.io/Time_series_mats/scripts/MebsInTime.html) and  [notebook](https://github.com/valdeanda/Time_series_mats/scripts/MebsInTime.ipynb)  formats. 

Run the notebook 

```bash
ipython3 notebook MebsInTime.ipynb
```
---

### `Barplot composition and logaritmic scale of taxonomic profile`

<img src="https://valdeanda.github.io/Time_series_mats/figures/Barplot.png" width="256" height="256" align="left">

Script available in  in [html](https://valdeanda.github.io/Time_series_mats/scripts/Bar_plots_composition.html) and [notebook](https://github.com/valdeanda/Time_series_mats/blob/master/scripts/Bar_plots_composition.ipynb) formats 

Run the notebook 

```bash
ipython3 notebook Bar_plots_composition.ipynb
```








# To generate the abundances profile from hmmsearch 
Use the following script that is also implemented in [MEBS software](https://github.com/eead-csic-compbio/metagenome_Pfam_score/tree/master/scripts)

```
 perl scripts/extract_pfam.pl -matrixdir matrix_pfam/ > data/test.pfams.tab 

```
